src/SpectrumTransformer.cpp

```diff
@@ -0,0 +1,409 @@
+/**********************************************************************
+
+Audacity: A Digital Audio Editor
+
+SpectrumTransformer.cpp
+
+Paul Licameli
+
+**********************************************************************/
+
+#include "SpectrumTransformer.h"
+
+#include <algorithm>
+#include "FFT.h"
+#include "WaveTrack.h"
+
+SpectrumTransformer::SpectrumTransformer( bool needsOutput,
+   eWindowFunctions inWindowType,
+   eWindowFunctions outWindowType,
+   size_t windowSize, unsigned stepsPerWindow,
+   bool leadingPadding, bool trailingPadding)
+: mWindowSize{ windowSize }
+, mSpectrumSize{ 1 + mWindowSize / 2 }
+, mStepsPerWindow{ stepsPerWindow }
+, mStepSize{ mWindowSize / mStepsPerWindow }
```

Comment:
Self note: Using int division to solve when the window size is not a multiple of the hop size.